apps/dashboard/test/models/motd_formatter/motd_formatter_markdown_erb_test.rb

```diff
@@ -1,88 +1,65 @@
-    path = "#{Rails.root}/test/fixtures/files/motd_valid_erb_md"
-    motd_file = MotdFile.new(path)
-    formatted_motd = MotdFormatterMarkdownErb.new(motd_file)
-    expected_file = OodAppkit.markdown.render("# Welcome to the Ohio Supercomputer Center!")
-    
-    assert_equal expected_file, formatted_motd.content
+    with_modified_env({ 'MOTD_FORMAT': "markdown_erb",'MOTD_PATH': "#{Rails.root}/test/fixtures/files/motd_valid_erb_md" }) do
+      expected_file = OodAppkit.markdown.render("# Welcome to the Ohio Supercomputer Center!")
+      assert_equal expected_file, MotdFile.new.formatter.content
+    end
   end
 
   test "motd-formatter-md-erb returns a standard error when given a invalid motd erb file" do
-    path = "#{Rails.root}/test/fixtures/files/motd_erb_standard_error"
-    motd_file = MotdFile.new(path)
-    
-    assert_raises(Exception) {
-      MotdFormatterMarkdownErb.new(motd_file)
-    }
+    with_modified_env({ 'MOTD_FORMAT': "markdown_erb",'MOTD_PATH': "#{Rails.root}/test/fixtures/files/motd_erb_standard_error" }) do
+      assert_raises(Exception) {
+        MotdFile.new.formatter
+      }
+    end
   end
 
   test "motd-formatter-md-erb returns an empty string when given an empty motd file" do
-    path = "#{Rails.root}/test/fixtures/files/motd_empty"
-    motd_file = MotdFile.new(path)
-    formatted_motd = MotdFormatterMarkdownErb.new(motd_file)
-
-    assert_equal '', formatted_motd.content
-  end
+    with_modified_env({ 'MOTD_FORMAT': "markdown_erb",'MOTD_PATH': "#{Rails.root}/test/fixtures/files/motd_empty" }) do
+      assert_equal '', MotdFile.new.formatter.content
+    end
+  end 
 
   test "motd-formatter-md-erb returns an empty string when given a missing file" do
-    path = "#{Rails.root}/test/fixtures/files/motd_missing"
-    motd_file = MotdFile.new(path)
-    formatted_motd = MotdFormatterMarkdownErb.new(motd_file)
-
-    assert_equal '', formatted_motd.content
-  end
-
-  test "motd-formatter-md-erb returns an empty stirng when given nill" do
-    motd_file = nil
-    formatted_motd = MotdFormatterMarkdownErb.new(motd_file)
-
-    assert_not_nil formatted_motd.content
+    with_modified_env({ 'MOTD_FORMAT': "markdown_erb",'MOTD_PATH': "#{Rails.root}/test/fixtures/files/motd_missing" }) do
+      assert_nil MotdFile.new.formatter
+    end
   end
 
   test 'content is html safe by default' do
-    path = "#{Rails.root}/test/fixtures/files/motd_md_erb_w_html"
-    motd = MotdFile.new(path)
-    formatted_motd = MotdFormatterMarkdownErb.new(motd)
-    content = formatted_motd.content
-
-    expected_content = "<h1>Some Markdown file</h1>\n" \
-    "\n" \
-    "var msg = 'this was a script';\n"
-
-    assert_not_nil(content)
-    assert_equal(expected_content, content)
+    with_modified_env({ 'MOTD_FORMAT': "markdown_erb",'MOTD_PATH': "#{Rails.root}/test/fixtures/files/motd_md_erb_w_html" }) do
+      expected_content = "<h1>Some Markdown file</h1>\n" \
+      "\n" \
+      "var msg = 'this was a script';\n"
```

Comment:
While we're at this, can you turn this into a HEREDOC just to make it a little cleaner?